### PR TITLE
[local_proxy]: install pg_session_jwt extension on demand

### PR DESCRIPTION
**Pull request by conradludgate**
Follow up on #9344. We want to install the extension automatically. We didn't want to couple the extension into compute_ctl so instead local_proxy is the one to issue requests specific to the extension.

depends on #9344 and #9395

### CODE REVIEW

File: proxy/src/serverless/backend.rs [hunk 1]
```diff
@@ -250,17 +254,51 @@ impl PoolingBackend {
...
+            let _permit = local_backend.initialize.acquire().await.unwrap();
+
+            // check again for race
+            if !self.local_pool.initialized(&conn_info) {
+                local_backend
+                    .compute_ctl
+                    .install_extension(&ExtensionInstallRequest {
+                        extension: EXT_NAME,
+                        database: conn_info.dbname.clone(),
+                        // todo: move to const or config
```
Comment (conradludgate) [inline proxy/src/serverless/backend.rs:276]:
```suggestion
```

File: proxy/src/serverless/backend.rs [hunk 2]
```diff
@@ -250,17 +254,51 @@ impl PoolingBackend {
...
+                        version: EXT_VERSION,
+                    })
+                    .await?;
+
+                local_backend
+                    .compute_ctl
+                    .grant_role(&SetRoleGrantsRequest {
+                        // fixed for pg_session_jwt
```
Comment (conradludgate) [inline proxy/src/serverless/backend.rs:284]:
```suggestion
```

File: proxy/src/compute_ctl/mod.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    ResonseError(#[source] reqwest::Error),
+}
+
+impl ComputeCtlApi {
+    pub async fn install_extension(
+        &self,
+        req: &ExtensionInstallRequest,
+    ) -> Result<ExtensionInstallResponse, ComputeCtlError> {
+        self.generic_request(req, Method::POST, |url| {
+            url.path_segments_mut().push("extensions");
```
Comment (conradludgate) [inline proxy/src/compute_ctl/mod.rs:60]: blocked until #9344 is merged

File: proxy/src/compute_ctl/mod.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        })
+        .await
+    }
+
+    pub async fn grant_role(
+        &self,
+        req: &SetRoleGrantsRequest,
+    ) -> Result<SetRoleGrantsResponse, ComputeCtlError> {
+        self.generic_request(req, Method::POST, |url| {
+            url.path_segments_mut().push("grants");
```
Comment (conradludgate) [inline proxy/src/compute_ctl/mod.rs:70]: blocked until #9395 is merged

File: proxy/src/compute_ctl/mod.rs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+pub enum ComputeCtlError {
+    #[error("connection error: {0}")]
+    ConnectionError(#[source] reqwest_middleware::Error),
+    #[error("request error [{status}]: {body:?}")]
+    RequestError {
+        status: StatusCode,
+        body: Option<GenericAPIError>,
+    },
+    #[error("response parsing error: {0}")]
+    ResonseError(#[source] reqwest::Error),
```
Comment (cloneable) [inline proxy/src/compute_ctl/mod.rs:51]: typo

File: proxy/src/serverless/backend.rs [hunk 6]
```diff
@@ -250,17 +254,49 @@ impl PoolingBackend {
...
-        let mut node_info = match &self.auth_backend {
+        let local_backend = match &self.auth_backend {
             auth::Backend::ControlPlane(_, ()) => {
                 unreachable!("only local_proxy can connect to local postgres")
             }
-            auth::Backend::Local(local) => local.node_info.clone(),
+            auth::Backend::Local(local) => local,
         };
 
+        #[allow(unreachable_code, clippy::todo)]
```
Comment (cloneable) [inline proxy/src/serverless/backend.rs:264]: Still needed?

**Review by cloneable — APPROVED** [review]

File: proxy/src/compute_ctl/mod.rs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            .await
+            .map_err(ComputeCtlError::ConnectionError)?;
+
+        let status = resp.status();
+        if status.is_client_error() || status.is_server_error() {
+            let body = resp.json().await.ok();
+            return Err(ComputeCtlError::RequestError { status, body });
+        }
+
+        resp.json().await.map_err(ComputeCtlError::ResonseError)
```
Comment (awarus) [inline proxy/src/compute_ctl/mod.rs:99]: Typo?

**Review by awarus — APPROVED** [review]

**Pull request merged**